cpp/daal/src/algorithms/kmeans/oneapi/kmeans_dense_lloyd_batch_kernel_ucapi_impl.i

```diff
@@ -426,19 +377,23 @@ const char * KMeansDenseLloydBatchKernelUCAPI<algorithmFPType>::getComputeSquare
 }
 
 template <typename algorithmFPType>
-void KMeansDenseLloydBatchKernelUCAPI<algorithmFPType>::computeSquares(ExecutionContextIface & context, const KernelPtr & kernel_compute_squares,
-                                                                       const Buffer<algorithmFPType> & data, UniversalBuffer & dataSq, uint32_t nRows,
+void KMeansDenseLloydBatchKernelUCAPI<algorithmFPType>::computeSquares(const Buffer<algorithmFPType> & data, UniversalBuffer & dataSq, uint32_t nRows,
                                                                        uint32_t nFeatures, Status * st)
 {
     DAAL_ITTNOTIFY_SCOPED_TASK(compute.computeSquares);
 
+    auto & context        = Environment::getInstance()->getDefaultExecutionContext();
```

Comment:
line 362: from `getComputeSquaresKernelName()` you are returning pointer to local object.

Reply:
All string literals are allocated at compile time.